cdc/sink/codec/config_test.go

```diff
@@ -17,127 +17,191 @@ import (
 	"net/url"
 	"testing"
 
-	"github.com/pingcap/tidb/util/timeutil"
 	"github.com/pingcap/tiflow/pkg/config"
-	cerror "github.com/pingcap/tiflow/pkg/errors"
 	"github.com/stretchr/testify/require"
 )
 
 func TestNewConfig(t *testing.T) {
 	t.Parallel()
-	c := NewConfig(config.ProtocolDefault, timeutil.SystemLocation())
-	require.Equal(t, c.protocol, config.ProtocolDefault)
-	require.Equal(t, c.maxMessageBytes, config.DefaultMaxMessageBytes)
-	require.Equal(t, c.maxBatchSize, defaultMaxBatchSize)
-	require.Equal(t, c.enableTiDBExtension, false)
-	require.Equal(t, c.avroRegistry, "")
+
+	c := NewConfig(config.ProtocolDefault)
+	require.Equal(t, config.ProtocolDefault, c.protocol)
+	require.Equal(t, config.DefaultMaxMessageBytes, c.maxMessageBytes)
+	require.Equal(t, defaultMaxBatchSize, c.maxBatchSize)
+	require.Equal(t, false, c.enableTiDBExtension)
+	require.Equal(t, "precise", c.avroDecimalHandlingMode)
+	require.Equal(t, "long", c.avroBigintUnsignedHandlingMode)
+	require.Equal(t, "", c.avroSchemaRegistry)
 }
 
 func TestConfigApplyValidate(t *testing.T) {
 	t.Parallel()
+
+	// enable-tidb-extension
 	uri := "kafka://127.0.0.1:9092/abc?protocol=canal-json&enable-tidb-extension=true"
 	sinkURI, err := url.Parse(uri)
-	require.Nil(t, err)
```

Comment:
what is the difference between `Nil` and `NoError` here, can we revert this change ?

Reply:
`NoError` is better. 1. It shows more semantics. 2. When fail, no error will show the trace of the error but `Nil` won't.